common/pulp/common/error_codes.py

```diff
@@ -122,6 +122,9 @@
 PLP0042 = Error("PLP0042", _("This request is forbidden."), [])
 PLP0043 = Error("PLP0043", _("Database 'write_concern' config can only be 'majority' or 'all'. "
                              "Refer to /etc/pulp/server.conf for proper use."), [])
+PLP0044 = Error("PLP0044", _("Content unit type '%(content_unit)s' explicitly defines an index for"
+                             "its unit key. This is not allowed because the platform handles it for"
+                             "you."), ['content_unit'])
```

Comment:
Actually, a ValueError kind of makes the most sense, because it's an incorrect value in the index dictionary.